prov/gni/src/gnix_rma.c

```diff
@@ -1452,14 +1446,28 @@ ssize_t _gnix_rma(struct gnix_fid_ep *ep, enum gnix_fab_req_type fr_type,
 
 	req->vc = vc;
 
+	/* Adding FI_FENCE to an FI_MORE list will break the FI_MORE Chain.
+	 * Current FI_MORE implementation does not create remote CQ events.
+	 * Remove FI_MORE flag when FI_FENCE or REMOTE EP requirements are
+	 * present. */
+
+	if ((flags & FI_FENCE) || (flags & FI_REMOTE_CQ_DATA) ||
+	    (req->vc->peer_caps & FI_RMA_EVENT)) {
+		flags &= ~FI_MORE;
+	}
+
+
 	/* Add reads/writes to slist when FI_MORE is present, Or
 	 * if this is the first message in the chain without FI_MORE.
 	 * When FI_MORE is not present, if the slists are not empty
-	 * it is the first message without FI_MORE. Do not add fence
-	 * reqs to the fi_more list. */
-	if ((flags & FI_MORE) || (!(flags & FI_FENCE) &&
-	   (!(slist_empty(&ep->more_write)) ||
-	    !(slist_empty(&ep->more_read))))) {
+	 * it is the first message without FI_MORE.
+	 * Do not add reqs with FI_FENCE or FI_REMOTE_CQ_DATA reqs to
+	 * the fi_more list. */
+	if ((flags & FI_MORE) ||
+	    (!(flags & FI_FENCE) && !(flags & FI_REMOTE_CQ_DATA) &&
+	     !(req->vc->peer_caps & FI_RMA_EVENT) &&
```

Comment:
How does this logic look to you @ztiffany   checks out with what we discussed above?
Formatting it well is weird though
```
 if ((flags & FI_MORE) ||
     (!(flags & FI_MORE) &&
     (!slist_empty(&ep->more_write) || !slist_empty(&ep->more_read)) &&
     !(flags & FI_FENCE || flags & FI_REMOTE_CQ_DATA || req->vc->peer_caps & FI_RMA_EVENT)))
```